### PR TITLE
 Integ tests: implement FSx Lustre tests

### PR DESCRIPTION
**Pull request by demartinofra**
- Test FSx Lustre correctly mounted on master node
- Test FSx Lustre correctly mounted on compute nodes
- Test ImportPath
- Test ExportPath

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/tests/test_fsx_lustre.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import boto3
+import pytest
+from retrying import retry
+
+from assertpy import assert_that
+from remote_command_executor import RemoteCommandExecutor
+from time_utils import minutes, seconds
+
+
+@pytest.mark.regions(["us-east-1", "eu-west-1", "cn-north-1", "us-gov-west-1"])
```
Comment (sean-smith) [inline tests/integration-tests/tests/test_fsx_lustre.py]: FSx isn't available in `cn-north-1` and `us-gov-west-1`. See https://aws.amazon.com/about-aws/global-infrastructure/regional-product-services/

Comment (demartinofra) [reply tests/integration-tests/tests/test_fsx_lustre.py]: Right, thanks! will update the regions to run only in commercial

File: tests/integration-tests/tests/test_fsx_lustre.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+from remote_command_executor import RemoteCommandExecutor
+from time_utils import minutes, seconds
+
+
+@pytest.mark.regions(["us-east-1", "eu-west-1"])
+@pytest.mark.instances(["c5.xlarge"])
+@pytest.mark.oss(["centos7"])
+@pytest.mark.schedulers(["sge"])
+@pytest.mark.usefixtures("os", "instance", "scheduler")
+def test_lustre(region, pcluster_config_reader, clusters_factory, s3_bucket_factory, test_datadir):
```
Comment (sean-smith) [inline tests/integration-tests/tests/test_fsx_lustre.py]: please rename to `test_fsx_lustre` or add an `fsx` marker

**Comment by sean-smith** [comment]
LGTM, just one small comment then 🚀

File: tests/integration-tests/tests/test_fsx_lustre.py (hunk 2 above)
Comment (demartinofra) [reply tests/integration-tests/tests/test_fsx_lustre.py]: Renamed!

File: tests/integration-tests/tests/test_fsx_lustre.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        "sudo lfs hsm_archive {mount_dir}/file_to_export && sleep 5".format(mount_dir=mount_dir)
+    )
+    remote_command_executor.run_remote_command(
+        "aws s3 cp s3://{bucket_name}/export_dir/file_to_export ./file_to_export".format(bucket_name=bucket_name)
+    )
+    result = remote_command_executor.run_remote_command("cat ./file_to_export")
+    assert_that(result.stdout).is_equal_to("Exported by FSx Lustre")
+
+
+def _assert_job_submitted(qsub_output):
```
Comment (enrico-usai) [inline tests/integration-tests/tests/test_fsx_lustre.py]: I've already seen this function in the awsbatch test. We should move this in the a utils module.

File: tests/integration-tests/tests/test_fsx_lustre.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    return match.group(1)
+
+
+@retry(retry_on_result=lambda result: result != 0, wait_fixed=seconds(7), stop_max_delay=minutes(5))
+def _wait_job_completed(remote_command_executor, job_id):
+    result = remote_command_executor.run_remote_command("qacct -j {0}".format(job_id), raise_on_error=False)
+    return result.return_code
+
+
+def _get_job_exit_status(remote_command_executor, job_id):
```
Comment (enrico-usai) [inline tests/integration-tests/tests/test_fsx_lustre.py]: same as before

File: tests/integration-tests/tests/test_fsx_lustre.py (hunk 3 above)
Comment (demartinofra) [reply tests/integration-tests/tests/test_fsx_lustre.py]: these functions are specific for the type of scheduler used (sge here, awsbatch in the other file). I was planning to move them to a common module only when actually multiple tests call them.

**Pull request merged**